src/nlsolve/nlsolve.jl

```diff
@@ -27,7 +27,7 @@ function nlsolve!(nlsolver::AbstractNLSolver, integrator)
       θ = ndz / ndzprev
 
       # divergence
-      if θ > 1
+      if θ > 1.2
```

Comment:
why this?

Reply:
https://github.com/LLNL/sundials/blob/38aa03b990e616db52acd053730c31fcc3eb91f9/src/cvode/cvode_nls.c#L27-L31

Sundials actually uses `2`, should we bump this?

Reply:
Yeah, we should probably adopt their "just keep trying a bit". Convergence of Newton is generally non-monotonic so assuming something close to it is doomed to extra factorizations.